meson.build

```diff
@@ -10,6 +10,49 @@ project(
   version : '0.1.0',
 )
 
+# project version
+version_parts = meson.project_version().split('.')
+version_major = version_parts[0]
+version_minor = version_parts[1]
+version_patch = version_parts[2]
+
+suffixed_version = meson.project_version()
+if get_option('version_suffix') != ''
+  suffixed_version += '-' + get_option('version_suffix')
+endif
+
+# library (API) version for linking
+#
+# rules:
+# sources changed: increment revision
+# binary interface changed: increment current, reset revision to 0
+#   binary interface changes backwards compatible?: increment age
+#   binary interface changes not backwards compatible?: reset age to 0
+library_revision = 0
+library_current = 1
+library_age = 0
+
+library_version = '@0@.@1@.@2@'.format(
+  library_current - library_age,
```

Comment:
Rather than replicating the libtool rules here, would it make sense to define this as "bump major and reset minor/patch for backward-incompatible API changes, bump minor and reset patch for backward-compatible API changes, bump patch for everything else"?  libtool has the odd property that the soversion can increase dramatically after breaking changes.

Reply:
I get very confused by ABI versioning. I reread the libtool pages about it just now, plus some stuff on SO, and I'm not any clearer about the best system, or how it might interact with darwin ABI version numbers.

I'm sure you know more about this than me, so I'll swap this for library_abi_major.minor.patch.